extensions/amp-accordion/1.0/accordion.js

```diff
@@ -0,0 +1,176 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {omit} from '../../../src/utils/object';
+import {sequentialIdGenerator} from '../../../src/utils/id-generator';
+import {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from '../../../src/preact';
+
+const AccordionContext = Preact.createContext(
+  /** @type {AccordionDef.ContextProps} */ ({})
+);
+
+/** @type {!Object<string, boolean>} */
+const EMPTY_EXPANDED_SET = {};
+
+const generateSectionId = sequentialIdGenerator();
+
+/**
+ * @param {!AccordionDef.Props} props
+ * @return {PreactDef.Renderable}
+ */
+export function Accordion({
+  as: Comp = 'section',
+  expandSingleSection = false,
+  children,
+  ...rest
+}) {
+  const [expandedSet, setExpandedSet] = useState(EMPTY_EXPANDED_SET);
+
+  useEffect(() => {
+    if (!expandSingleSection) {
+      return;
+    }
+    setExpandedSet((expandedSet) => {
+      const newExpandedSet = {};
+      let expanded = 0;
+      for (const k in expandedSet) {
+        newExpandedSet[k] = expandedSet[k] && expanded++ == 0;
+      }
+      return newExpandedSet;
+    });
+  }, [expandSingleSection]);
+
+  const registerSection = useCallback((id) => {
+    return () => setExpandedSet((expandedSet) => omit(expandedSet, id));
+  }, []);
+
+  const toggleExpanded = useCallback(
+    (id, defaultExpanded) => {
+      setExpandedSet((expandedSet) => {
+        const newValue = !isExpanded(
+          id,
+          defaultExpanded,
+          expandedSet,
+          expandSingleSection
+        );
+        let newExpandedSet;
+        if (newValue && expandSingleSection) {
+          newExpandedSet = {[id]: newValue};
+          for (const k in expandedSet) {
+            if (k != id) {
+              newExpandedSet[k] = false;
+            }
+          }
+        } else {
+          newExpandedSet = {...expandedSet, [id]: newValue};
+        }
+        return newExpandedSet;
+      });
+    },
+    [expandSingleSection]
+  );
+
+  const context = useMemo(
+    () =>
+      /** @type {!AccordionDef.ContextProps} */ ({
+        registerSection,
+        toggleExpanded,
+        isExpanded: (id, defaultExpanded) =>
+          isExpanded(id, defaultExpanded, expandedSet, expandSingleSection),
+      }),
+    [expandedSet, expandSingleSection, registerSection, toggleExpanded]
+  );
+
+  return (
+    <Comp {...rest}>
+      <AccordionContext.Provider value={context}>
+        {children}
+      </AccordionContext.Provider>
+    </Comp>
+  );
+}
+
+/**
+ * @param {string} id
+ * @param {boolean} defaultExpanded
+ * @param {!Object<string, boolean>} expandedSet
+ * @param {boolean} expandSingleSection
+ * @return {boolean}
+ */
+function isExpanded(id, defaultExpanded, expandedSet, expandSingleSection) {
+  const current = expandedSet[id];
+  if (current != null) {
```

Comment:
`x != null` is the same as `(x !== undefined && x !== null)`. Since `null` is not one of the possible values here, this expression is the same as just `x !== undefined`. But it's a little bit shorter in the resulting binary code. Too cryptic?

Reply:
Oh neat, thanks for explaining. Feel free to leave as is.